mscore/metaedit.cpp

```diff
@@ -29,95 +30,263 @@ namespace Ms {
 //---------------------------------------------------------
 
 MetaEditDialog::MetaEditDialog(Score* s, QWidget* parent)
-   : QDialog(parent)
+   : QDialog(parent),
+     m_score(s),
+     m_dirty(false)
       {
-      setObjectName("MetaEditDialog");
+      if (objectName().isEmpty())
+            setObjectName("MetaEditDialog");
+
       setupUi(this);
-      setWindowFlags(this->windowFlags() & ~Qt::WindowContextHelpButtonHint);
-      score = s;
-      dirty = false;
+      QDialog::setWindowFlag(Qt::WindowContextHelpButtonHint, false);
+      QDialog::setWindowFlag(Qt::WindowMinMaxButtonsHint);
+
+      QString mscoreVrs = m_score->mscoreVersion();
+      if (!mscoreVrs.isEmpty())
+            version->setText(mscoreVrs);
 
-      level->setValue(score->mscVersion());
-      version->setText(score->mscoreVersion());
-      int rev = score->mscoreRevision();
+      int scoreVrs = m_score->mscVersion();
+      if (scoreVrs)
+            level->setText(QString::number(scoreVrs, 10)); // in decimal
+
+      int rev = m_score->mscoreRevision();
       if (rev > 99999)  // MuseScore 1.3 is decimal 5702, 2.0 and later uses a 7-digit hex SHA
             revision->setText(QString::number(rev, 16));
       else
             revision->setText(QString::number(rev, 10));
-      filePath->setText(score->importedFilePath());
-      filePath->setTextInteractionFlags(Qt::TextSelectableByMouse);
 
-      int idx = 0;
+      QString sfilePath(m_score->importedFilePath());
+      if (!sfilePath.isEmpty())
```

Comment:
Should I remove those unnecessary ifs? (line 45, 49, 59, one will be added on 53.)

Reply:
I am not sure what is the rule for that. Should we always check just in case?